field_friend/localization/gnss.py

```diff
@@ -116,33 +129,50 @@ def _on_rtk_fix(self) -> None:
             yaw=yaw,
             time=self.current.timestamp)
         self.observed_poses.append(pose)
-
-    async def update_robot_pose(self) -> None:
-        assert not self.is_paused
-        if self.ensure_gnss:
-            while len(self.observed_poses) < self.needed_poses:
-                if rosys.time() - self.last_pose_update < self.min_seconds_between_updates:
-                    return
-                await rosys.sleep(0.1)
-        if not self.observed_poses:
-            return
+        
+        if len(self.observed_poses) > self.needed_poses: # Ensure fresh data
+            self.observed_poses.pop(0)
+        
+        if not self._is_paused:
+            self._update_robot_pose()
+
+    async def wait_for_robot_pose_update(self, timeout: Optional[float] = None) -> None:
```

Comment:
```suggestion
    async def wait_for_robot_pose_update(self, timeout: float | None = None) -> None:
```
The [styling guide](https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945) calls for `| None` instead of `Optional` for Python 3.10+ and we want to gradually follow that